modules/core/src/core/animation-loop.js

```diff
@@ -101,49 +101,54 @@ export default class AnimationLoop {
 
   // Starts a render loop if not already running
   // @param {Object} context - contains frame specific info (E.g. tick, width, height, etc)
-  start(opts = {}) {
+  async start(opts = {}) {
     if (this._running) {
-      return this;
+      return;
     }
     this._running = true;
+
     // console.debug(`Starting ${this.constructor.name}`);
     // Wait for start promise before rendering frame
-    this._startPromise = getPageLoadPromise()
-      .then(() => {
-        if (!this._running || this._initialized) {
-          return null;
-        }
-
-        // Create the WebGL context
-        this._createWebGLContext(opts);
-        this._createFramebuffer();
-        this._startEventHandling();
-
-        // Initialize the callback data
-        this._initializeCallbackData();
-        this._updateCallbackData();
-
-        // Default viewport setup, in case onInitialize wants to render
-        this._resizeCanvasDrawingBuffer();
-        this._resizeViewport();
-
-        // Note: onIntialize can return a promise (in case it needs to load resources)
-        const initializationPromise = this.onInitialize(this.animationProps);
-        this._initialized = true;
-        return initializationPromise;
-      })
-      .then(appContext => {
-        if (this._running) {
-          this._addCallbackData(appContext || {});
-          if (appContext !== false) {
-            this._startLoop();
-          }
-        }
-      });
+    await getPageLoadPromise();
+
+    if (!this._running || this._initialized) {
+      return;
+    }
+
+    // Create the WebGL context
+    this._createWebGLContext(opts);
+    this._createFramebuffer();
+    this._startEventHandling();
+
+    // Initialize the callback data
+    this._initializeCallbackData();
+    this._updateCallbackData();
+
+    // Default viewport setup, in case onInitialize wants to render
+    this._resizeCanvasDrawingBuffer();
+    this._resizeViewport();
+
+    // Note: onIntialize can return a promise (in case it needs to load resources)
+    const initializationPromise = this.onInitialize(this.animationProps);
+    this._initialized = true;
+
+    const appContext = await initializationPromise;
+
+    if (this._running) {
+      this._addCallbackData(appContext || {});
+      if (appContext !== false) {
+        this._startRenderLoop();
+      }
+    }
+  }
+
+  // Stops a render loop if already running, finalizing
+  stop() {
+    this._endRenderLoop();
     return this;
```

Comment:
Since we've updated `start()` to no longer return `this`, should we update `stop` to also not return `this`?